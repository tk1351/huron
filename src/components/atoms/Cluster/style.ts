import { css } from '@emotion/react'
import { JustifyContent } from '../../../constants/types'

type ClusterProps = {
  justifyContent: JustifyContent
  gap: string
}

export const cluster = ({ justifyContent, gap }: ClusterProps) =>
  css({
    display: 'flex',
    justifyContent,
    gap: `${gap}`
  })
