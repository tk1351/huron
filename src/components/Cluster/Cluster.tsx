import { FC } from 'react'
import { cluster } from './style'
import { JustifyContent } from '../../constants/types'

type Props = {
  justifyContent?: JustifyContent
  gap?: string
  children: JSX.Element | JSX.Element[]
}

export const Cluster: FC<Props> = ({
  justifyContent = 'flex-start',
  gap = '0px',
  children
}) => <div css={cluster({ justifyContent, gap })}>{children}</div>
