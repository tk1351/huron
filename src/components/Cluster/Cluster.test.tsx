import * as renderer from 'react-test-renderer'
import { Cluster } from './Cluster'

const Component = (
  <Cluster>
    <div>App</div>
    <div>App2</div>
  </Cluster>
)

describe('Testing Cluster Component', () => {
  describe('Render', () => {
    it('正しくClusterコンポーネントが表示されていること', () => {
      const App = renderer.create(Component).toJSON()
      expect(App).toMatchSnapshot()
    })
  })
})
