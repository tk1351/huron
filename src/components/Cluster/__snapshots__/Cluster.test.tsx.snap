// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Cluster Component Render 正しくClusterコンポーネントが表示されていること 1`] = `
<div
  css={
    Object {
      "map": undefined,
      "name": "14ggwnl",
      "next": undefined,
      "styles": "display:flex;justify-content:flex-start;gap:0px;",
      "toString": [Function],
    }
  }
>
  <div>
    App
  </div>
  <div>
    App2
  </div>
</div>
`;
